x-pack/plugins/lists/server/routes/read_privileges_route.mock.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+interface Cluster {
+  monitor_ml: boolean;
+  manage_ccr: boolean;
+  manage_index_templates: boolean;
+  monitor_watcher: boolean;
+  monitor_transform: boolean;
+  read_ilm: boolean;
+  manage_security: boolean;
+  manage_own_api_key: boolean;
+  manage_saml: boolean;
+  all: boolean;
+  manage_ilm: boolean;
+  manage_ingest_pipelines: boolean;
+  read_ccr: boolean;
+  manage_rollup: boolean;
+  monitor: boolean;
+  manage_watcher: boolean;
+  manage: boolean;
+  manage_transform: boolean;
+  manage_api_key: boolean;
+  manage_token: boolean;
+  manage_ml: boolean;
+  manage_pipeline: boolean;
+  monitor_rollup: boolean;
+  transport_client: boolean;
+  create_snapshot: boolean;
+}
+
+interface Index {
+  [indexName: string]: {
+    all: boolean;
+    manage_ilm: boolean;
+    read: boolean;
+    create_index: boolean;
+    read_cross_cluster: boolean;
+    index: boolean;
+    monitor: boolean;
+    delete: boolean;
+    manage: boolean;
+    delete_index: boolean;
+    create_doc: boolean;
+    view_index_metadata: boolean;
+    create: boolean;
+    manage_follow_index: boolean;
+    manage_leader_index: boolean;
+    write: boolean;
+  };
+}
+
+interface IndexPrivilege {
+  application: {};
+  cluster: Cluster;
+  has_all_requested: boolean;
+  index: Index;
+  username: string;
+}
+
+export interface Privilege {
+  listItems: IndexPrivilege;
+  lists: IndexPrivilege;
+  is_authenticated: boolean;
+}
+
+export const getReadPrivilegeMock = (
+  listIndex: string = '.lists-default',
```

Comment:
nit: I'm starting to see these "lists index" strings strewn around in various forms; even the config itself uses its own copy. We should probably move these to `common/constants` at some point sooner than later.

Reply:
Which string? I didn't see this one being used everywhere `'.lists-default'` just yet? But we have a mocks constant file that I can move this to. That would make the most sense. I will do it for this PR.

Reply:
I think my initial reaction was: how will a developer know what the default value is if we keep using slightly different strings everywhere? However, after giving it some more thought I realize that that is a Bad Take™. The user can configure the index name to be (almost) any string that they want, so testing against the same string everywhere would be an opportunity for bugs to crop up.